doc/developers/advanced_installation.rst

```diff
@@ -279,9 +279,9 @@ path environment variable.
 -------------
 
 for 32-bit python it is possible use the standalone installers for
-`microsoft visual c++ express 2008 <http://go.microsoft.com/?linkid=7729279>`_
+`microsoft visual c++ express 2008 <http://download.microsoft.com/download/A/5/4/A54BADB6-9C3F-478D-8657-93B3FC9FE62D/vcsetup.exe>`_
 for python 2 or
-`microsoft visual c++ express 2010 <http://go.microsoft.com/?linkid=9709949>`_
+`microsoft visual c++ express 2010 <https://www.visualstudio.com/products/visual-studio-dev-essentials-vs>`_
```

Comment:
If we cannot find an official direct download link from microsoft anymore I would remove the link...